src/name_service/delivery/delete.clj

```diff
@@ -0,0 +1,34 @@
+(ns name-service.delivery.delete
+  (:require [cheshire.core :as json]
+            [clojure.spec.alpha :as s]
+            [name-service.delivery.http :as http]
+            [name-service.delivery.spec :as spec]
+            [name-service.delivery.state :as state]
+            [name-service.delivery.utils :as utils]))
+
+
+(defn- delete-keymap [kv]
+  (let [key (utils/map-key-first kv)
+        value (utils/map-value-first kv)]
+    (utils/action->response (state/delete-keymap key value))))
+
+(defn- delete-by-key [key-delete kv]
```

Comment:
Those argument names are not readable. Check Elements of Clojure's chapter on Naming I sent on the Slack channel.
Better names would be something like: `key-to-delete` and `where-cond` or `where-kv`. Optionally, leave `key-delete` as `key` but add docstring to explain meaning of arguments.